src/use-visitor-data.ts

```diff
@@ -73,16 +73,20 @@ export function useVisitorData<TExtended extends boolean>(
         setState((state) => (state.isLoading ? { ...state, isLoading: false } : state))
       }
     },
-    [getOptions, getVisitorData]
+    [state.getOptions, getVisitorData]
   )
 
   useEffect(() => {
-    if (immediate && (!previousGetOptions || !deepEquals(getOptions, previousGetOptions))) {
+    if (immediate) {
       getData().catch((error) => {
         console.error(`Failed to fetch visitor data on mount: ${error}`)
       })
     }
-  }, [immediate, getData, previousGetOptions, getOptions])
+  }, [immediate, getData])
+
+  if (!deepEquals(state.getOptions, getOptions)) {
+    setState((state) => ({ ...state, getOptions }))
+  }
```

Comment:
```suggestion
  if (state.getOptions != getOptions && !deepEquals(state.getOptions, getOptions)) {
    setState((state) => ({ ...state, getOptions }))
  }
```

nit: We can compare references first, just to avoid calling `deepEquals` on every render. If the references are equal, it means that the objects are the same, so there is no need for deep check.

In most cases, it won't matter because customers usually pass the options inline:
```  
const { isLoading, error, getData } = useVisitorData(
    { ignoreCache: true },
    { immediate: false }
  )
```
then the reference will change on every render. It will only have impact if the `getOptions` are memoized
 or stored outside of the component:


```
const options = { ignoreCache: true }

function Component() {
  const { isLoading, error, getData } = useVisitorData(
    options,
    { immediate: false }
  )
  // ...
}
```

Still, in that case it should offer at least slight performance improvement :)